dashboard/app/controllers/api_controller.rb

```diff
@@ -504,6 +504,63 @@ def section_assessments
     render json: data
   end
 
+  # For each assessment in a script, return an object of script_level IDs to question data.
+  # Question data includes the question text, all possible answers, and the correct answers.
+  # TODO(caleybrock): currently only used in internal experiment, must add controller tests.
+  def assessments_structure
+    script = load_script
+
+    # Only authorized teachers have access to locked question and answer data.
+    render status: :forbidden unless current_user.authorized_teacher?
+
+    level_group_script_levels = script.script_levels.includes(:levels).where('levels.type' => 'LevelGroup')
+
+    assessments = {}
+
+    level_group_script_levels.map do |script_level|
+      next unless script_level.long_assessment?
+
+      # Don't allow somebody to peek inside an anonymous survey using this API.
+      next if script_level.anonymous?
+
+      # The actual level group that corresponds to the script_level
+      level_group = script_level.levels[0]
+
+      # A structure to store the individual level structure in the level_group
+      questions = {}
+
+      # For each level in the multi group (ignore pages structure information)
+      level_group.levels.each do |level|
+        # A single level corresponds to a single question
+        question = {}
+
+        case level
+        when FreeResponse
+          question = {
+            type: level.type,
```

Comment:
I prefer to make use of polymorphism than switch on class type. The code to create these hashes could live entirely in the `Multi` and `FreeResponse` classes, and this loop would just be:
```
level_group.levels.each do |level|
  questions[level.id] = level.question_summary
end
```

Reply:
I'll follow up on this.